src/ErrorCodes.php

```diff
@@ -0,0 +1,380 @@
+    const IndexBuildAlreadyInProgress = 285;
+    const ChangeStreamHistoryLost = 286;
+    const TransactionCoordinatorDeadlineTaskCanceled = 287;
+    const ChecksumMismatch = 288;
+    const WaitForMajorityServiceEarlierOpTimeAvailable = 289;
+    const TransactionExceededLifetimeLimitSeconds = 290;
+    const NoQueryExecutionPlans = 291;
+    const QueryExceededMemoryLimitNoDiskUseAllowed = 292;
+    const InvalidSeedList = 293;
+    const InvalidTopologyType = 294;
+    const InvalidHeartBeatFrequency = 295;
+    const TopologySetNameRequired = 296;
+    const HierarchicalAcquisitionLevelViolation = 297;
+    const InvalidServerType = 298;
+    const OCSPCertificateStatusRevoked = 299;
+    const RangeDeletionAbandonedBecauseCollectionWithUUIDDoesNotExist = 300;
+    const DataCorruptionDetected = 301;
+    const OCSPCertificateStatusUnknown = 302;
+    const SplitHorizonChange = 303;
+    const ShardInvalidatedForTargeting = 304;
+    const ReadThroughCacheKeyNotFound = 305;
+    const ReadThroughCacheLookupCanceled = 306;
+    const RangeDeletionAbandonedBecauseTaskDocumentDoesNotExist = 307;
+    const CurrentConfigNotCommittedYet = 308;
+    const SocketException = 9001;
+    const OBSOLETE_RecvStaleConfig = 9996;
+    const CannotGrowDocumentInCappedNamespace = 10003;
+    const NotMaster = 10107;
+    const BSONObjectTooLarge = 10334;
+    const DuplicateKey = 11000;
+    const InterruptedAtShutdown = 11600;
+    const Interrupted = 11601;
+    const InterruptedDueToReplStateChange = 11602;
+    const BackgroundOperationInProgressForDatabase = 12586;
+    const BackgroundOperationInProgressForNamespace = 12587;
+    const OBSOLETE_PrepareConfigsFailed = 13104;
+    const MergeStageNoMatchingDocument = 13113;
+    const DatabaseDifferCase = 13297;
+    const StaleConfig = 13388;
+    const NotMasterNoSlaveOk = 13435;
+    const NotMasterOrSecondary = 13436;
+    const OutOfDiskSpace = 14031;
+    const OSBELETE_KeyTooLong = 17280;
+    const ClientMarkedKilled = 46841;
+
+    /**
+     * Checking if instance of error exception is related to mongodb.
+     * examples:
+     * try{
+     *     // mongodb commands ...  
+     * }catch(\Exception $e){
+     *     if(ErrorCodes::is($e))
+     *         echo 'this is a mongodb error.';
+     * }
+     * 
+     * try{
+     *     // mongodb commands ...  
+     * }catch(\Exception $e){
+     *     if(ErrorCodes::is($e,ErrorCodes::WriteConflict))
+     *         echo 'this is a mongodb write conflict error.';
+     * }
+     * @param \Exception $e instance of error exception.
+     * @param int|null $verifyCode if set then returning true value if error code exists.
+     * @return bool returning true value when this instance is related to mongodb.
+     * when you set $verifyCode parameter then returns true value if instance is related to mongodb and error code exists.
+    */
+    public static function is($e, $verifyCode = null){
+        if(
```

Comment:
Conditions could be inverted for early return:

```php
if (
  !$e instanceof MongoDBExceptionInterface &&
  (!$e instanceof YiiMongoDBException || !$e->getPrevious() instanceof MongoDBExceptionInterface)
) {
    return false;
}

if ($verifyCode !== null && $e->getCode() !== $verifyCode) {
    return false;
}

return true;
```

Reply:
A variety of status:
1. `$e` is  instance of `MongoDBExceptionInterface`.✔
2. `$e` is instance of `YiiMongoDBException` and previous exception **is not** instance of `MongoDBExceptionInterface `.✖
3. `$e` is instance of `YiiMongoDBException` and previous exception **is** instance of `MongoDBExceptionInterface `.✔
4. `$e` is another instance.✖
5. a parent of instance `$e` is `MongoDBExceptionInterface`. (I'm thinking about it)

we need to `1` and `3`. if status is `1` or `3` then return `true` else return `false` without checking `$verifyCode`;
your codes does not work on this status.